### PR TITLE
Splited report into two tables

### PR DESCRIPTION
**Pull request by bond95**
# Description

For now it's just consume report and save it into two tables.

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

## Testing steps

Run tests

### CODE REVIEW

File: migration/mig_0013_add_rule_table.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import (
+	"database/sql"
+
+	"github.com/RedHatInsights/insights-results-aggregator/types"
+)
+
+var mig0013AddRuleTable = Migration{
+	StepUp: func(tx *sql.Tx, driver types.DBDriver) error {
+		_, err := tx.Exec(`
+		CREATE TABLE rule (
```
Comment (tisnik) [inline migration/mig_0013_add_rule_table.go:30]: Probably it might be good to think about better name of this table.

File: storage/storage.go [hunk 2]
```diff
@@ -424,14 +440,57 @@ func (storage DBStorage) WriteReportForCluster(
...
+	kafkaOffset types.KafkaOffset,
+) error {
+	// Skip writing the report if it isn't newer than a report
+	// that is already in the database for the same cluster.
+	if oldLastChecked, exists := storage.clustersLastChecked[clusterName]; exists && !lastCheckedTime.After(oldLastChecked) {
+		return types.ErrOldReport
+	}
+
 	// Check if there is a more recent report for the cluster already in the database.
-	rows, err := tx.Query(
```
Comment (tisnik) [inline storage/storage.go:428]: don't removing this operation from transaction cause prohibition of atomic upsert operation?

**Comment by quarckster** [comment]
@tisnik ack

File: migration/mig_0013_add_rule_table.go (hunk 1 above)
Comment (bond95) [reply migration/mig_0013_add_rule_table.go:30]: Maybe `hit_rule`?

Comment (tisnik) [reply migration/mig_0013_add_rule_table.go:30]: or `rule_hit`, what you prefer

File: storage/storage.go (hunk 2 above)
Comment (bond95) [reply storage/storage.go:428]: I didn't quite understand your question.

Comment (tisnik) [reply storage/storage.go:428]: I mean if the query is called outside transaction, there's no guarantee you can trust it's resultset later. Some other process can update database in meantime.

Comment (bond95) [reply storage/storage.go:428]: hmm... ok, it makes sense

File: consumer/processing.go [hunk 3]
```diff
@@ -211,35 +212,41 @@ func checkReportStructure(r Report) error {
 }
 
 // parseMessage tries to parse incoming message and read all required attributes from it
-func parseMessage(messageValue []byte) (incomingMessage, error) {
+func parseMessage(messageValue []byte) (incomingMessage, []types.ReportItem, error) {
```
Comment (joselsegura) [inline consumer/processing.go:215]: Why not including "rules" as a field of the `incomingMessage` type instead changing the number of returned valus, and in fact the rules array is part of the incoming message

**Comment by quarckster** [comment]
@bond95 please rebase

**Pull request closed without merging**